src/utils/notifications.js

```diff
@@ -67,84 +71,94 @@ const validCollectibleTransaction = (transaction: ?Object): boolean => {
     if (!parsedNotification || !validBcxTransaction(parsedNotification)) return result;
-
-    let message = '';
-    let title = '';
-    const {
-      asset,
-      status,
-      value,
-      decimals,
-    } = parsedNotification;
-    const sender = parsedNotification.fromAddress.toUpperCase();
-    const receiver = parsedNotification.toAddress.toUpperCase();
-    const amount = utils.formatUnits(EthersBigNumber.from(value.toString()), decimals);
-
-    if (receiver === myEthAddress && status === 'pending') {
-      title = t('tokenValue', { value: amount, token: asset });
-      message = t('notification.received');
-    } else if (receiver === myEthAddress && status === 'confirmed') {
-      title = `${amount} ${asset}`;
-      message = t('notification.transactionConfirmed');
-    } else if (sender === myEthAddress && status === 'pending') {
-      title = t('tokenValue', { value: amount, token: asset });
-      message = t('notification.transactionSent');
-    } else if (sender === myEthAddress && status === 'confirmed') {
-      title = t('tokenValue', { value: amount, token: asset });
-      message = t('notification.transactionReceived');
-    }
-
-    result = {
-      title,
-      message,
-      asset,
-      status,
-      type: notification.type,
-    };
+    result = { type: notification.type };
   }
 
   if (notification.type === BADGE) {
-    result = {
-      title: t('notification.success'),
-      message: t('notification.newBadgeReceived'),
-      type: notification.type,
-    };
+    result = { type: notification.type };
   }
 
   if (!!notification.type && notification.type.toUpperCase() === COLLECTIBLE) {
     if (!parsedNotification || !validCollectibleTransaction(parsedNotification)) return result;
+    result = { type: COLLECTIBLE };
+  }
+
+  return result;
+};
+
+const isSameAddress = (a: ?string, b: ?string) => a && b && a.toUpperCase() === b.toUpperCase();
+
+export const getToastNotification = (data: mixed, myEthAddress: string): null | Notification => {
+  if (typeof data !== 'object') return null;
+  const { type, msg } = data ?? {};
+  const notification = typeof msg === 'string' && parseNotification(msg);
+  if (!notification) return null;
+
+  if (type === FCM_DATA_TYPE.BCX) {
+    if (!validBcxTransaction(notification)) return null;
 
-    let message = '';
-    let title = '';
     const {
-      contractName,
-    } = parsedNotification;
-    const sender = parsedNotification.fromAddress.toUpperCase();
-    const receiver = parsedNotification.toAddress.toUpperCase();
-
-    if (receiver === myEthAddress) {
-      title = contractName;
-      message = t('notification.receivedCollectible');
-    } else if (sender === myEthAddress) {
-      title = contractName;
-      message = t('notification.collectibleSentAndReceived');
+      asset,
+      status,
+      value,
+      decimals,
+      fromAddress,
+      toAddress,
+    } = notification;
+
+    const paymentInfo = t('tokenValue', {
```

Comment:
Nit: I believe that key "tokenValue" or "transactionValue", or simply "value" would be more readable and understandable (as a translation prop) for those who would translate it
(as if, for example, you only see `{{paymentInfo}} is coming shortly`, it might be hard to understand what this is about whereas `{{value}} is coming shortly` is clearer. Don't you think?)

Reply:
Sure, will fix. (It was based on https://github.com/pillarwallet/pillarwallet/blob/develop/src/actions/smartWalletActions.js#L706)